### ISSUE TITLE
Regarding to obtaining  F1, precision and recall from generate_ROC.py 

### ISSUE THREAD

**Issue by cervantes-loves-ai** [ekosman/AnomalyDetectionCVPR2018-Pytorch#19]
from tpr and fpr how to obtain  F1, precision and recall? thanks in advance.

**Comment by ekosman** [comment]
Hi @cervantes-loves-ai,

These are two lists with values corresponding to the threshold values. Thus you have 2 options:
1. You firstly specify a threshold value and then for each element decide whether it belongs to the positive class or the negative class. After that, you can use library functions like sklearn.metrics.recall_score for computing the desired metric.
2. For a pair of tpr, fpr you can calculate TP,FP,FN, TN by:
- tp = tpr * p
- fp = fp * n
- tn = (1-fpr) * n
- fn = (1 - tpr) * p

There you have all the values for recomputing the f1-score or our metrics as you want, using the original formulas.

**Comment by cervantes-loves-ai** [comment]
thank you for your quick reply

one thing
what will be positive samples and negative samples in generate_ROC.py?

**Comment by ekosman** [comment]
Positives are frames that contain an anomaly
